### PR TITLE
Automated generation and publishing of contract API docs with solidity-docgen

### LINKED ISSUE #383: [DOC] Integrate solidity-docgen and jsdoc for dev branch
# Summary

In order to make the https://docs.superfluid.finance site an one-stop shop for our developers, we would like to host our reference docs for both `ethereum-contracts` and `js-doc` projects, generated by [solidity-docgen](https://github.com/OpenZeppelin/solidity-docgen) and [jsdoc](https://github.com/jsdoc/jsdoc).

# Technical Description

(To be finalized)

1. Our doc site is a gitbook project available at https://github.com/superfluid-finance/superfluid-protocol-docs
2. Integrate solidity-docgen and jsdoc as part of [ci-canary](https://github.com/superfluid-finance/protocol-monorepo/blob/dev/.github/workflows/ci-canary.yml) workflow.
3. Generated docs should be to hosted and referenced in superfluid docs site nicely (how?)

# Bounty Info

Bounty Prize: TBD

Bonus question:

- How about the `subgraph` reference docs?

### PR DESCRIPTION
**Pull request by d10r**
First PR for https://github.com/superfluid-finance/protocol-monorepo/issues/383

Changes:
* added a hardhat plugin for automated doc generation with [solidity-docgen](https://github.com/OpenZeppelin/solidity-docgen) - since v0.6 (rewrite) of that package, the hardhat plugin is the main (only?) way to operate it
* added handlebar templates to get better (hopefully) results (layout, formatting...)
* adapted natspec docs in interface files to yield better results, e.g. by using `@custom` tags recently introduced

There's still some work to be done, will be documented in the ticket

### CODE REVIEW

**Comment by saflamini** [comment]
@d10r 

On the titles for each function and event, what if we format:

- Functions with a () appended to the end? "createFlow()" instead of "fn createFlow" ?
- Events with 'Event' appended to the end of the description? "flowUpdated Event" instead of "Event flowUpdated"

One other macro point: there's a *lot* of room for improvement in our interface comments. Many of the functions lack descriptions of the function and their associated return values/parameters.
Improving the quality of our comments in each interface file is an effort that I can help lead.

File: .github/workflows/handler.update-evm-contracts-docs.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - uses: actions/checkout@v3
+        with:
+          repository: superfluid-finance/build-scripts
+          path: build-scripts
+
+      - name: Create the docs
+        run: |
+          yarn workspace @superfluid-finance/ethereum-contracts docgen
+
+# Code to upload to refs.superfluid.finance. The md files aren't useful there, need to render first
```
Comment (hellwolf) [inline .github/workflows/handler.update-evm-contracts-docs.yml:41]: do you want to keep these as comments or clean it up?

File: yarn.lock [hunk 2]
```diff
@@ -11582,7 +11582,7 @@ gzip-size@^6.0.0:
   dependencies:
     duplexer "^0.1.2"
 
-handlebars@^4.0.1, handlebars@^4.7.6:
```
Comment (hellwolf) [inline yarn.lock:11585]: these seems suspicious, could use `npm ls handlebars` to checkout how these dependencies came about.

not a big deal, but worth cleanup if easy.

**Review by hellwolf — APPROVED** [review]
looking good

**Comment by hellwolf** [comment]
@d10r shoudl we merge?

File: .github/workflows/handler.update-evm-contracts-docs.yml (hunk 1 above)
Comment (saflamini) [reply .github/workflows/handler.update-evm-contracts-docs.yml:41]: @hellwolf this is up to @d10r, but I am comfortable with removing

**Pull request merged**